src/rust/inetstack/protocols/layer2/mod.rs

```diff
@@ -92,7 +91,15 @@ impl SharedLayer2Endpoint {
         Ok(batch)
     }
 
-    pub fn transmit(
+    pub fn transmit_arp_packet(&mut self, remote_link_addr: MacAddress, pkt: DemiBuffer) -> Result<(), Fail> {
```

Comment:
Why are these functions needed? It looks like they are just switching the `EtherType2`. If the aim is to bring clarity, another way to think about this is to move the `EtherType2` parameter first in the `transmit()` fn, so it's evident to the reader what's happening. Then, only `transmit()` can be made public.

Reply:
It is so that I can remove a dependency on EtherType2 from the higher layer. I thought this would be easier to not mess up, then to accidentally send a packet of the wrong type. Let me know what you think.

Reply:
Makse sense, we can keep it the way you have it now.